shared/pkg/scanner/trivy/scanner.go

```diff
@@ -198,6 +199,32 @@ func (a *Scanner) Run(sourceType utils.SourceType, userInput string) error {
 			return
 		}
 
+		switch sourceType {
+		case utils.DOCKERARCHIVE:
+			trivyOptions.ImageOptions.Input = userInput
+		case utils.OCIARCHIVE:
+			tmpDir, err := os.MkdirTemp("", "")
+			if err != nil {
+				a.setError(fmt.Errorf("unable to create temp directory: %w", err))
+				return
+			}
+			defer func() {
+				err := os.RemoveAll(tmpDir)
+				if err != nil {
+					a.logger.WithError(err).Errorf("unable to remove temp directory %s", tmpDir)
+				}
+			}()
+			err = utilsTrivy.UntarToDirectory(userInput, tmpDir)
+			if err != nil {
+				a.setError(fmt.Errorf("unable to untar %s to temp directory: %w", userInput, err))
+				return
+			}
+
+			trivyOptions.ImageOptions.Input = tmpDir
+		case utils.IMAGE, utils.ROOTFS, utils.DIR, utils.FILE, utils.SBOM:
+			// Nothing to do here
+		}
+
```

Comment:
same logic as in the analyzer, why not putting it into a function?

Reply:
+1 I'll refactor